python4j/python4j-core/src/main/java/org/eclipse/python4j/PythonGC.java

```diff
@@ -0,0 +1,115 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Konduit K.K.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+
+package org.eclipse.python4j;
+
+import org.bytedeco.cpython.PyObject;
+
+import java.io.Closeable;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.bytedeco.cpython.global.python.*;
+
+public class PythonGC implements Closeable{
+
+    private PythonGC previousFrame = null;
+    private boolean active = true;
+    private static PythonGC currentFrame = new PythonGC();
+
+    private Set<PyObject> objects = new HashSet<>();
```

Comment:
Is this safe? We're relying on equals and hashCode methods inherited from Pointer...
hashCode uses address, so that should be fine.
Is it possible to have two PyObject java objects pointing to the same python object/address? If so - this might not work, looking at Pointer.equals, and things like alreadyRegistered might not work as expected.

Not sure TBH but worth checking carefully.

Reply:
`Pointer.equals` also uses address:

```java
   public boolean equals(Object obj) {
        if (obj == this) {
            return true;
        } else if (obj == null) {
            return this.isNull();
        } else if (obj.getClass() != this.getClass() && obj.getClass() != Pointer.class && this.getClass() != Pointer.class) {
            return false;
        } else {
            Pointer other = (Pointer)obj;
            return this.address == other.address && this.position == other.position;
        }
    }

```